### DISCUSSION TITLE
Proposal: Add Support for Custom Fields on Tags

### DISCUSSION THREAD

**Discussion by st3nzel** [netbox-community/netbox#17846] [Ideas]
I propose the addition of custom fields for Tags in NetBox, similar to the existing custom field support for other objects (e.g., Devices, Sites, Cables). This feature would provide more flexibility in organizing and managing metadata for tags, enhancing how they are used for tagging and categorizing objects in NetBox.

Use Case
Tags in NetBox are already an incredibly powerful feature for organizing and filtering objects across various models, such as Devices, Cables, IP Addresses, etc. However, there are scenarios where storing additional metadata directly on tags would be highly beneficial:

Enhanced Tag Metadata:
Tags are often used to represent logical groupings or physical pathways (e.g., cable routes, network segments). Custom fields would allow users to store more information directly on tags, such as utilization thresholds, physical properties, or other contextual information specific to that tag.

I look forward to the community's feedback on this proposal!

**Comment by jeremystretch** [comment]
Sounds like FR #16080.
